akka-stream/src/main/scala/akka/stream/javadsl/RetryFlow.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.javadsl
+
+import java.util.Optional
+
+import akka.annotation.ApiMayChange
+import akka.japi.Pair
+import akka.stream.scaladsl
+import akka.util.JavaDurationConverters._
+
+import scala.compat.java8.OptionConverters._
+
+object RetryFlow {
+
+  /**
+   * API may change!
+   *
+   * Allows retrying individual elements in the stream with an exponential backoff.
+   *
+   * The retry condition is controlled by the `decideRetry` function. It takes the originally emitted
+   * element with its context, and the response emitted by `flow`, and may return a request to be retried.
+   *
+   * The implementation of the `RetryFlow` requires that `flow` follows one-in-one-out semantics,
+   * the [[akka.stream.javadsl.FlowWithContext FlowWithContext]] may not filter elements,
+   * nor emit more than one element per incoming element.
```

Comment:
Note here as well that only one element will be in the nested flow. Also: note what happens (or doesn't happen) if it does filter or emits more than one.